src/BotWebApi/Controllers/VacanciesController.cs

```diff
@@ -14,10 +15,123 @@ public class VacanciesController : ApiController
     {
         IBotContext _context = new DummyBotContext();
 
-        public string GetVacancies()
+        [HttpGet]
+        public HttpResponseMessage All()
         {
-            var JsonVacancies = JsonConvert.SerializeObject(_context.Vacancies);
-            return JsonVacancies;
+            return new HttpResponseMessage()
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(JsonConvert.SerializeObject(_context.Vacancies, Formatting.Indented, new JsonSerializerSettings
+                {
+                   
+
+                })),
+            };
+        }
+
+        [HttpGet]
+        public HttpResponseMessage Get(int id)
+        {
+            HttpResponseMessage response;
+            if (_context.Vacancies.Any(x => x.Id == id))
+            {
+                var foundedVacancy = _context.Vacancies.First(x => x.Id == id);
+                response = new HttpResponseMessage()
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(JsonConvert.SerializeObject(foundedVacancy, Formatting.Indented, new JsonSerializerSettings
+                    {
+                        
+                    }))
+                };
+            }
+            else
+            {
+                response = new HttpResponseMessage(HttpStatusCode.NotFound);
+            }
+            return response;
+        }
+
+        [Route("api/vacancies/{vacancyId}/candidates")]
+        [HttpGet]
+        public HttpResponseMessage VacanciesProgress(int vacancyId)
+        {
+            HttpResponseMessage response;
+            if (_context.Vacancies.Any(x => x.Id == vacancyId))
+            {
+                var foundedVacancy = _context.Vacancies.First(x => x.Id == vacancyId);
+                response = new HttpResponseMessage()
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(JsonConvert.SerializeObject(foundedVacancy.CandidatesProgress, Formatting.Indented, new JsonSerializerSettings
+                    {
+                        
+                    }))
+                };
+            }
+            else
+            {
+                response = new HttpResponseMessage(HttpStatusCode.NotFound);
+            }
+            return response;
         }
+
+
+        [HttpDelete]
+        public HttpResponseMessage Delete(int id)
+        {
+            HttpResponseMessage response;
+            var foundedVacancy = _context.Vacancies.First(x => x.Id == id);
```

Comment:
this doesn't returns a null pointer, just an exception.
so use FirstOrDefault instead.